File: examples/go/search_1.go
package main

import "fmt"

func linearSearch(arr []int, target int) int {
	for i, v := range arr {
		if v == target {
			return i
		}
	}
	return -1
}

func main() {
	arr := []int{3, 7, 2, 9, 1, 5}
	fmt.Println(linearSearch(arr, 9))
	fmt.Println(linearSearch(arr, 4))
}

File: examples/rust/search_1.rs
fn linear_search(arr: &[i32], target: i32) -> Option<usize> {
    for (i, &x) in arr.iter().enumerate() {
        if x == target {
            return Some(i);
        }
    }
    None
}

fn main() {
    let data = [3, 7, 2, 9, 1];
    println!("{:?}", linear_search(&data, 7));
    println!("{:?}", linear_search(&data, 5));
}
